client/src/components/Draftail/sources/ModalWorkflowSource.js

```diff
@@ -23,16 +23,31 @@ export const getChooserConfig = (entityType, entity, selectedText) => {
 
   switch (entityType.type) {
   case ENTITY_TYPE.IMAGE:
+    if (entity) {
+      const data = entity.getData();
+      url = `${global.chooserUrls.imageChooser}${data.id}/select_format/`;
+      urlParams = {
+        format: data.format,
+        alt_text: data.alt,
+      };
+    } else {
+      url = `${global.chooserUrls.imageChooser}?select_format=true`;
+      urlParams = {};
+    }
     return {
-      url: `${global.chooserUrls.imageChooser}?select_format=true`,
-      urlParams: {},
+      url,
+      urlParams,
       onload: global.IMAGE_CHOOSER_MODAL_ONLOAD_HANDLERS,
     };
 
   case EMBED:
+    urlParams = {};
+    if (entity) {
+      urlParams.url = entity.getData().url;
+    }
```

Comment:
We need tests for this and above in `ModalWorkflowSource.test.js`.

Reply:
Tests added here
https://github.com/wagtail/wagtail/pull/5885/files#diff-36d75bb432538146bbccff1e4daffa3cR42-R52
and here
https://github.com/wagtail/wagtail/pull/5885/files#diff-36d75bb432538146bbccff1e4daffa3cR62-R69